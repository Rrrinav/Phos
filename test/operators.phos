// 1. Arithmetic
print("Arithmetic:");
print(10 + -5);
print(10.5 - 15.5);
print(-5 * -2);
print(10.0 / 4.0);
print(10 % 3);

// 2. String Concatenation
print("\nString Concat:");
print("Phos " + "Language");

// 3. Comparison
print("\nComparison:");
print(5 > 3);
print("a" != "b");
print("a" == "b");
print("a" == "a");

// 4. Logical
print("\nLogical:");
print(true && (false || true));
print(!false);

// 5. Precedence and Casting
let result : f64 = (2 + 3) * (10 - 6) / 2; // 5 * 4 / 2 = 10
print("\nPrecedence (5 * 4 / 2): " + (result as string));
print("Cast to string: " + (42 as string));
