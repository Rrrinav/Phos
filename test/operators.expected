Arithmetic:
5
-5.000000
10
2.500000
1

String Concat:
Phos Language

Comparison:
true
true
false
true

Logical:
true
true

Precedence (5 * 4 / 2): 10.000000
Cast to string: 42
